src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1765,6 +1765,7 @@ because xbuild doesn't support framework reference assemblies.
   <ItemGroup>
     <_GeneratedAndroidEnvironment Include="mono.enable_assembly_preload=0" Condition=" '$(AndroidEnablePreloadAssemblies)' != 'True' " />
     <_GeneratedAndroidEnvironment Include="DOTNET_MODIFIABLE_ASSEMBLIES=Debug" Condition=" '$(AndroidIncludeDebugSymbols)' == 'true' and '$(AndroidUseInterpreter)' == 'true' " />
+    <_GeneratedAndroidEnvironment Include="DOTNET_DiagnosticPorts=$(DiagnosticConfiguration)" Condition=" '$(DiagnosticConfiguration)' != '' " />
```

Comment:
Any risk this might get stuck in the build or would a new build and/or run without any Diagnostic* properties set remove all diagnostic support in the build making sure this part of the build gets updated.

Reply:
This part looks OK for incremental builds, this target has no Inputs/Outputs, so it always runs.

Then `<WriteLinesToFile/>` uses `WriteOnlyWhenDifferent="True"`, which would update the timestamp on the `@(AndroidEnvironment)` file it writes if changed.

Later on, it looks like all the important targets will rerun if `@(AndroidEnvironment)` files change.